### DISCUSSION TITLE
defaults write params

### DISCUSSION THREAD

**Discussion by mummifiedclown** [aonez/Keka#1110] [General]
Just wondering if there is a list of all the settings that can be set via the defaults write command.  Looking to do some automation and some aspects of Keka's prefs are a little hard to work with on a file basis.

**Comment by aonez** [comment]
There's no such thing, only a [list with obscure settings](https://github.com/aonez/Keka/wiki/Hidden-configuration) that can only be modified with the Terminal.
You can see all the set values with:

```
defaults read com.aone.keka
```

But probably you already know that. If you need any specific setting let me know.

**Reply by mummifiedclown** [reply]
That’s a good start - thanks.  I don’t suppose you could share how you’re generating the data payloads in the Bookmarks array?  I know it’s a base64 encoding of binary/text data that includes the uuid of the APFS data volume and some version of the path, but that’s as far as I’ve gotten.

**Reply by aonez** [reply]
The bookmarks can only be created within the app since they use sandbox APIs from Apple. I'm pretty sure those are not portable or reproducible outside the app (as a sandbox security mesure).

**Reply by mummifiedclown** [reply]
That makes sense - too bad there’s not a non-sandboxed version of Keka that can just use the Full Disk Access or Files and Folders privacy prefs.  Thanks for the insight.

**Reply by aonez** [reply]
While there’s no un-sandboxed release you can do it pretty easily:

https://github.com/aonez/Keka/wiki/Rar-compression#un-sandbox-keka
